packages/find-workspace-dir/src/index.ts

```diff
@@ -9,9 +10,17 @@ export default async function findWorkspaceDir (cwd: string) {
   const workspaceManifestDirEnvVar = process.env[WORKSPACE_DIR_ENV_VAR] ?? process.env[WORKSPACE_DIR_ENV_VAR.toLowerCase()]
   const workspaceManifestLocation = workspaceManifestDirEnvVar
     ? path.join(workspaceManifestDirEnvVar, 'pnpm-workspace.yaml')
-    : await findUp([WORKSPACE_MANIFEST_FILENAME, 'pnpm-workspace.yml'], { cwd })
+    : await findUp([WORKSPACE_MANIFEST_FILENAME, 'pnpm-workspace.yml'], { cwd: await getRealPath(cwd) })
   if (workspaceManifestLocation?.endsWith('.yml')) {
     throw new PnpmError('BAD_WORKSPACE_MANIFEST_NAME', `The workspace manifest file should be named "pnpm-workspace.yaml". File found: ${workspaceManifestLocation}`)
   }
   return workspaceManifestLocation && path.dirname(workspaceManifestLocation)
 }
+
+async function getRealPath (path: string) {
+  return new Promise<string>((resolve) => {
+    fs.realpath.native(path, function (err, resolvedPath) {
```

Comment:
Why shoud `fs.realpath.native` be used instead of `fs.realpath`?

Reply:
`fs.realpath.native` return the actual casing of a path as it is on disk  https://github.com/nodejs/node/pull/15776/files#diff-05bc37150ff4e7aa51dc7a28c71164a9e715f45e323ac9a5c67e061821aad3d2R8, `fs.realpath` is  resolving `..`, `.` and sym links

Reply:
in that case it is better to add a comment about it because in the Node.js docs this information is not given.

Reply:
@zkochan I've added comments on `getRealPath`